checker/src/org/checkerframework/checker/nullness/NullnessVisitor.java

```diff
@@ -424,16 +442,33 @@ public Void visitTypeCast(TypeCastTree node, Void p) {
     // ///////////// Utility methods //////////////////////////////
 
     /**
-     * Issues a 'dereference.of.nullable' if the type is not of a {@link NonNull} type.
+     * Issues the error message if the type is not of a {@link NonNull} type.
      *
      * @param tree the tree where the error is to reported
      * @param errMsg the error message (must be {@link CompilerMessageKey})
+     * @return whether or not the check succeeded
      */
-    private void checkForNullability(ExpressionTree tree, /*@CompilerMessageKey*/ String errMsg) {
+    private boolean checkForNullability(
+            ExpressionTree tree, /*@CompilerMessageKey*/ String errMsg) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(tree);
+        return checkForNullability(type, tree, errMsg);
+    }
+
+    /**
+     * Issues a 'dereference.of.nullable' if the type is not of a {@link NonNull} type.
```

Comment:
Shouldn't this first sentence be changed as the version above, i.e. replace the 'dereference.of.nullable' by 'the error message'.
Some indication of what the difference between the two methods is would also be good.